### PR TITLE
junos: Added configure_user

### PR DESCRIPTION
**Pull request by NilashishC**
- Added junos_user_manager action plugin
- Added configure_user task and j2 template
- Added integration tests
- Usage:
```
---
- name: Test for user_account_management role
  hosts: junos
  vars:
    users:
      - name: ansible
        role: super-user
        encrypted_password: "{{ 'testpassword' | password_hash('sha512') }}"

      - name: new_user_1
        full_name: Testuser
        role: read-only
        sshkey: "{{ lookup('file', '~/.ssh/id_rsa.pub') }}"
    
  tasks:
    - import_role:
        name: juniper_junos
        tasks_from: configure_user
```

### CODE REVIEW

File: templates/configure_user.j2 [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+{% else %}
+{% if user.role is defined %}
+set system login user {{ user.name }} class {{ user.role }}
+{% endif %}
+{% if user.full_name is defined %}
+set system login user {{ user.name }} full-name {{ user.full_name }}
+{% endif %}
+{% if user.sshkey is defined %}
+{% set key_type = user.sshkey.split(' ')[0] %}
+set system login user {{ user.name }} authentication {{ key_type }} "{{ user.sshkey }}"
```
Comment (ganeshrn) [inline templates/configure_user.j2:16]: Can setting encrypted password be added as part of this PR. Refer https://github.com/ansible/ansible/pull/44660/files

**Review by ganeshrn — CHANGES_REQUESTED** [review]
Please add integration tests for this fetaure

File: templates/configure_user.j2 (hunk 1 above)
Comment (NilashishC) [reply templates/configure_user.j2:16]: ACK. Added this.

File: tests/configure_user/configure_user/tasks/configure_user.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+---
+- debug: msg="START configure_user function on connection={{ ansible_connection }}"
+
+- name: ensure netconf is enabled
+  junos_netconf:
+    netconf_port: 22
```
Comment (ganeshrn) [inline tests/configure_user/configure_user/tasks/configure_user.yml]: Please change this to `netconf_port: "{{ netconf_port }}"

File: tests/configure_user/test.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+- hosts: appliance
+  connection: network_cli
+  roles:
+    - configure_user
```
Comment (ganeshrn) [inline tests/configure_user/test.yml:4]: Please pass netconf_port varaible here
```
  roles:
    - config_manager
  vars:
    netconf_port: 830
```

Comment (NilashishC) [reply tests/configure_user/test.yml:4]: Fixed.

File: tests/configure_user/configure_user/tasks/configure_user.yml (hunk 2 above)
Comment (NilashishC) [reply tests/configure_user/configure_user/tasks/configure_user.yml]: Done.

**Review by ganeshrn — APPROVED** [review]

**Comment by ganeshrn** [comment]
recheck

**Comment by ganeshrn** [comment]
recheck

**Comment by samccann** [comment]
looks good from the docs perspective.

**Pull request merged**